docs/dev/spec-capability.md

```diff
@@ -0,0 +1,67 @@
+
+The keeper provides no other core functionality that can be found in other modules, such as queriers, REST and CLI handlers, and the genesis state.
+
+## Initialization
+
+When you initialize the application, the keeper must be instantiated with a persistent store key and an ephemeral in-memory store key.
+
+```
+type App struct {
+  // ...
+
+  capabilityKeeper *capability.Keeper
+}
+
+func NewApp(...) *App {
+  // ...
+
+  app.capabilityKeeper = capability.NewKeeper(codec, persistentStoreKey, memStoreKey)
+}
+```
+
+After the keeper is created, it can create scoped subkeepers, which are passed to other modules that can create, authenticate, and claim capabilities. After all the necessary scoped subkeepers are created and the state is loaded, you must initialize and seal the main capability keeper to populate the ephemeral in-memory state and to prevent more scoped subkeepers from being created.
+
+```
+func NewApp(...) *App {
+  // ...
+
+  // Initialize and seal the capability keeper so all persistent capabilities
+  // are loaded in-memory and prevent any further modules from creating scoped
+  // sub-keepers.
+  ctx := app.BaseApp.NewContext(true, tmproto.Header{})
+  app.capabilityKeeper.InitializeAndSeal(ctx)
+
+  return app
+}
+```
+
+## Concepts
+
+Capabilities are multi-owner. A scoped subkeeper can create a capability via `NewCapability,` which creates a unique, unforgeable, object-capability reference. The newly created capability is automatically persisted. The calling module does not need to call `ClaimCapability`. Calling `NewCapability` creates the capability with the calling module and name as a tuple to be the capabilities first owner.
+
+Capabilities can be claimed by other modules, which add them as owners. `ClaimCapability` allows a module to claim a capability key that it has received from another module so that `GetCapability` calls made later will succeed. If a module that receives a capability wants to access it by name later, `ClaimCapability` must be called. Because capabilities are multi-owner, if multiple modules have one capability reference, all the modules own it. If a module receives a capability from another module but does not call `ClaimCapability`, it may use it in the executing transaction but will not be able to access it afterward.
+
+Any module can call `AuthenticateCapability` to check whether a capability corresponds to a particular name, including un-trusted user input, with which the calling module previously associated it.
+
+`GetCapability` allows a module to fetch a capability that it has previously claimed by name. The module is not allowed to retrieve capabilities that it does not own.
+
+### Stores
```

Comment:
Section doesn't adhere to the conventional format of the rest, but we can fix this in a later iteration